src/sentry/api/endpoints/user_authenticator_details.py

```diff
@@ -159,7 +161,18 @@ def delete(self, request: Request, user, auth_id, interface_device_id=None) -> R
                 user, ignore_backup=True
             )
             last_2fa_method = len(enrolled_methods) == 1
-            require_2fa = user.get_orgs_require_2fa().exists()
+            # TODO(hybrid-cloud): synchronize flags for org settings onto org mapping to make this RPC unnecessary?
```

Comment:
To be honest, as written, this RPC can never actually work, unless the flags are on the org mapping.  There is some unscoped work that involves moving this logic to the org mapping table instead of querying orgs directly, because that will be impossible.

So, yeah, this is another thing we must investigate, broadly.